share/spice/coupons/coupons.js

```diff
@@ -0,0 +1,88 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_coupons = function(api_result){
+        // Validate the response (customize for your Spice)
+        if (!api_result || api_result.error) {
+            return Spice.failed('coupons');
+        }
+
+        var tabname, translation_ValidUntil, translation_Unknown;
+        switch (api_result.locale) {
```

Comment:
I think it would be better to have an object defined outside our callback function, with the keys being the locales. Then we don't need to iterate over all locales because we can just do something like `var locale_data = locales[locale]`